x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -607,63 +610,76 @@ export class TaskRunner<
       (err: ElasticsearchError) => executionStatusFromError(err, runDate)
     );
 
+    // New consolidated statuses for lastRun
+    const { lastRun, metrics: executionMetrics } = map<
+      RuleTaskStateAndMetrics,
+      ElasticsearchError,
+      ILastRun
+    >(
+      stateWithMetrics,
+      (ruleRunStateWithMetrics) => lastRunFromState(ruleRunStateWithMetrics),
+      (err: ElasticsearchError) => lastRunFromError(err)
+    );
+
     if (apm.currentTransaction) {
       if (executionStatus.status === 'ok' || executionStatus.status === 'active') {
         apm.currentTransaction.setOutcome('success');
       } else if (executionStatus.status === 'error' || executionStatus.status === 'unknown') {
         apm.currentTransaction.setOutcome('failure');
+      } else if (lastRun.outcome === 'succeeded') {
```

Comment:
nit: since we're calculating both executionStatus and lastRun every time, seems we could remove the checks against executionStatus and just use the lastRun to determine apm current transaction outcome?

Reply:
We did that initially but we decided to have both just to be a bit more defensive haha. Eventually we would deprecate any references to `executionStatus`. Although I can remove the checks to `executionStatus` here if we feel strongly about it.